lib/bambora/rest_client.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'faraday'
+
+module Bambora
+  class RestClient
+    attr_accessor :base_url, :merchant_id, :sub_merchant_id
+
+    # Initialze a client that makes REST requests. The RestClient is used by the JSONClient.
+    #
+    # @example
+    #   rest_client = Bambora::RestClient.new(
+    #     base_url: ENV.fetch('BAMBORA_BASE_URL'),
+    #     merchant_id: ENV.fetch('BAMBORA_MERCHANT_ID'),
+    #     sub_merchant_id: ENV.fetch('BAMBORA_SUB_MERCHANT_ID'),
+    #   )
+    def initialize(options = {})
+      if !options[:version].nil? && options[:version].upcase != 'V1'
```

Comment:
Not necessarily blocking, but this piece of code is now in a few places. Is it possible to turn this into a shareable piece of code across the gem?